### PR TITLE
Feature/inverse btn

### PR DESCRIPTION
**Pull request by RobStallion**
# Description

# Checklist

Pull request contains:

- [ ] A new component
- [x] Component maintenance: improvement / bug fix / etc
- [ ] Component library change: storybook / webpack / etc

Definition of done:

- [ ] Includes theme changes for both Uswitch and money
- [ ] Work has been tested in multiple browsers
- [ ] PR description includes description of change
- [ ] PR description includes screenshot of change
- [ ] If new component, designer has approved screenshot
- [ ] If the change will affect other teams, that team knows about this change
- [ ] If introducing a new component behaviour, added a story to cover that case.
- [ ] If component change, version updated

### CODE REVIEW

File: src/elements/button/src/index.tsx [hunk 1]
```diff
@@ -1,14 +1,43 @@
...
+const invertTheme = (theme: any, variant: any = {}) => {
+  const backgroundColor = variant.color
+  const color = variant.backgroundColor || 'white'
+  const borderColor = color
+
+  const hoverColor = theme.colors[backgroundColor]
+    ? darken(theme.colors[backgroundColor], 0.1)()
+    : null
+
+  console.log({ variant })
```
Comment (jonathan-fielding) [inline src/elements/button/src/index.tsx]: left over console.log

File: src/elements/button/src/index.tsx [hunk 2]
```diff
@@ -1,14 +1,43 @@
...
+
+  return {
+    ...variant,
+    backgroundColor,
+    color,
+    borderColor,
+    borderWidth: 2,
+    borderStyle: 'solid',
```
Comment (jonathan-fielding) [inline src/elements/button/src/index.tsx]: should we not add the variant after so it overrides

**Comment by callumacrae** [comment]
There's a missing dependency (this is also why the build is failing):

image

I don't think this approach will work for the money theme, where we won't want the colours to be just inverted for the invert button - we'd want the border colour to come from the background, and the background colour to be white:

image

The inverse primary button in storybook at the moment looks a bit strange:

image

File: src/elements/button/src/index.tsx (hunk 2 above)
Comment (RobStallion) [reply src/elements/button/src/index.tsx]: @jonathan-fieldin I wanted to ensure that these took priority over the default variant.

**Pull request merged**